presto-main/src/main/java/com/facebook/presto/server/remotetask/HttpLocationFactory.java

```diff
@@ -87,6 +89,30 @@ public URI createTaskLocation(InternalNode node, TaskId taskId)
                 .build();
     }
 
+    @Override
+    public URI createRemoteTaskLocation(InternalNode node, TaskId taskId)
+    {
+        requireNonNull(node, "node is null");
+        requireNonNull(taskId, "taskId is null");
+
+        if (taskCommunicationProtocol.equals(CommunicationProtocol.HTTP)) {
+            return createTaskLocation(node, taskId);
+        }
+
+        OptionalInt thriftPort = node.getThriftPort();
+
+        HttpUriBuilder builder = uriBuilderFrom(node.getInternalUri());
+        if (taskCommunicationProtocol.equals(CommunicationProtocol.THRIFT) && thriftPort.isPresent()) {
+            builder.scheme("thrift");
+            builder.port(thriftPort.getAsInt());
+        }
+        else {
+            // fall back to http case
```

Comment:
This fall back is a bit fragile because you have to always make sure the following matches `createTaskLocation`, which is not as obvious. Should we actually consider just calling `createTaskLocation`?

Reply:
I would love to but we cannot merge the two `createTaskLocation` into one before we have completely implemented thrift version for `/v1/task` endpoint. For shuffle in `/v1/task`, it will rely on `createRemoteTaskLocation` but for scheduling in `/v1/task`, it will use `createTaskLocation`. Because scheduling path does not support thrift, which implies it will fail on `thrift://` scheme.